bigframes/ml/compose.py

```diff
@@ -130,8 +225,10 @@ def camel_to_snake(name):
             if "transformSql" not in transform_col_dict:
                 continue
             transform_sql: str = transform_col_dict["transformSql"]
-            if not transform_sql.startswith("ML."):
-                continue
+
+            # workaround for bug in bq_model returning " AS `...`" suffix for flexible names
+            if cls.AS_FLEXNAME_SUFFIX_RX.match(transform_sql):
+                transform_sql = cls.AS_FLEXNAME_SUFFIX_RX.match(transform_sql).group(1)
```

Comment:
Getting some mypy errors here. I think because mypy isn't smart enough to see the null check here. Let's try this instead:

```suggestion
            flex_name_match = cls.AS_FLEXNAME_SUFFIX_RX.match(transform_sql)
            if flex_name_match:
                transform_sql = flex_name_match.group(1)
```

```
nox > mypy bigframes tests/system tests/unit --check-untyped-defs --explicit-package-bases '--exclude="^third_party"'
[bigframes/ml/compose.py:231](https://cs.corp.google.com/piper///depot/google3/bigframes/ml/compose.py?l=231): error: Item "None" of "Match[str] | None" has no attribute "group"  [union-attr]
Found 1 error in 1 file (checked 235 source files)
nox > Command mypy bigframes tests/system tests/unit --check-untyped-defs --explicit-package-bases '--exclude="^third_party"' failed with exit code 1
nox > Session mypy failed.
```